### ISSUE TITLE
Editor: Unnatural state of save confirmation panel when none are checked

### ISSUE THREAD

**Issue by t-hamano** [WordPress/gutenberg#52689]
## What problem does this address?
If you try to perform a save action after updating a navigation, template part, synchronization pattern, etc., you will see the following panel:

![save_panel_checked](https://github.com/WordPress/gutenberg/assets/54422211/874a0022-52f6-44a1-baaa-8a4c797b67a2)

If none are checked, there appears to be a small padding below the "Are you ready to save?" text. Some users may also wonder why the Save button cannot be pressed.

![save_panel_unchecked](https://github.com/WordPress/gutenberg/assets/54422211/1d380f0f-c987-445f-8d5c-815f1d5648d3)

This problem is similar to the modal.

![modal](https://github.com/WordPress/gutenberg/assets/54422211/1172159b-f3d5-4a5f-a124-f6a236c51c84)

## What is your proposed solution?

How about letting users know what they need to do to save it, as follows:

![alert](https://github.com/WordPress/gutenberg/assets/54422211/b4d322cd-4d7e-4b71-b866-c9b88abc40bd)

**Comment by benwormald** [comment]
Maybe something that explains why the save button is unavailable, in addition to some instructive text?
 
<img width="218" alt="Screenshot 2023-08-24 at 11 47 56 AM" src="https://github.com/WordPress/gutenberg/assets/8432342/d4234903-5e07-40a0-a3e1-15e291ba5ff3">
